utils/fixed_decimal/src/decimal.rs

```diff
@@ -388,28 +400,40 @@ impl FixedDecimal {
     /// let mut dec = FixedDecimal::from(42);
     /// assert_eq!("42", dec.to_string());
     ///
-    /// dec.multiply_pow10(3).expect("Bounds are small");
+    /// dec.multiply_pow10(3);
     /// assert_eq!("42000", dec.to_string());
     /// ```
-    pub fn multiply_pow10(&mut self, delta: i16) -> Result<(), Error> {
+    pub fn multiply_pow10(&mut self, delta: i16) {
         match delta.cmp(&0) {
             Ordering::Greater => {
-                self.upper_magnitude = self
-                    .upper_magnitude
-                    .checked_add(delta)
-                    .ok_or(Error::Limit)?;
-                // If we get here, then the magnitude change is in-bounds.
-                let lower_magnitude = self.lower_magnitude + delta;
-                self.lower_magnitude = cmp::min(0, lower_magnitude);
+                let upper_magnitude = self.upper_magnitude.checked_add(delta);
+                match upper_magnitude {
+                    Some(upper_magnitude) => {
+                        self.upper_magnitude = upper_magnitude;
+                        // If we get here, then the magnitude change is in-bounds.
+                        let lower_magnitude = self.lower_magnitude + delta;
+                        self.lower_magnitude = cmp::min(0, lower_magnitude);
+                    }
+                    None => {
+                        // there is an overflow
+                        self.clear();
+                    }
```

Comment:
Issue: We shouldn't `clear`; I think we should drop the digits that would overflow. For example, `102030` with multiply_pow10 that goes out of range should produce something like `020300000000...`

Reply:
I do not agree for two reasons:

1. This an extreme case that the users should not reach. and instead of Error, we are going to clear the number, so the user can understand that there is a problem happened.
2. It will take coding effort in order to to comply with this extreme case.

I suggest that we submit this PR with this behaviour and discuss if this behaviour is acceptable or not. if it is not, we can easily change the internal behaviour.

Reply:
Okay, that's reasonable. At least the API will be the same, and we can discuss the exceptional behavior in another issue. Could you please open an issue for that?

Reply:
done https://github.com/unicode-org/icu4x/issues/2297